pkg/controller/operators/catalog/operator.go

```diff
@@ -281,15 +281,29 @@ func (o *Operator) handleDeletion(obj interface{}) {
 }
 
 func (o *Operator) handleCatSrcDeletion(obj interface{}) {
-	if catsrc, ok := obj.(*v1alpha1.CatalogSource); ok {
-		sourceKey := resolver.CatalogKey{Name: catsrc.GetName(), Namespace: catsrc.GetNamespace()}
-		func() {
-			o.sourcesLock.Lock()
-			defer o.sourcesLock.Unlock()
-			delete(o.sources, sourceKey)
-		}()
-		o.Log.WithField("source", sourceKey).Info("removed client for deleted catalogsource")
+	catsrc, ok := obj.(metav1.Object)
+	if !ok {
+		if !ok {
+			tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+			if !ok {
+				utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+				return
+			}
+
+			catsrc, ok = tombstone.Obj.(metav1.Object)
+			if !ok {
+				utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not a Namespace %#v", obj))
+				return
+			}
+		}
 	}
+	sourceKey := resolver.CatalogKey{Name: catsrc.GetName(), Namespace: catsrc.GetNamespace()}
+	func() {
```

Comment:
why does this need to execute in an anonymous function if the outer function returns immediately after this (ie. No real work is being done between the lock and the return)?

Reply:
For future changes, this keeps the unlock close to to the reason we're locking.

Reply:
I understand that - was just commenting on this specific case - it makes sense to be consistent with the rest of the codebase.